plugin-api/src/main/java/net/clgd/ccemux/api/emulation/filesystem/VirtualDirectory.java

```diff
@@ -91,15 +86,45 @@ public VirtualDirectory() {
 		this.children = new HashMap<>();
 	}
 
-	public boolean hasEntry(String name) {
+	public boolean hasEntry(@Nonnull String name) {
 		return children.containsKey(name);
 	}
 
-	public VirtualMountEntry getEntry(String name) {
+	@Nullable
+	public VirtualMountEntry getEntry(@Nonnull String name) {
 		return children.get(name);
 	}
 
+	@Nonnull
 	public Set<String> getEntryNames() {
 		return children.keySet();
 	}
+
+	@Override
+	@SuppressWarnings("all")
+	public boolean equals(final java.lang.Object o) {
+		if (o == this) return true;
+		if (!(o instanceof VirtualDirectory)) return false;
+		final VirtualDirectory other = (VirtualDirectory) o;
+		if (!other.canEqual((java.lang.Object) this)) return false;
+		final java.lang.Object this$children = this.children;
```

Comment:
`lang.Object`. Also `this$children` feels … blughr.

Reply:
I'm tempted to just use Kotlin and data classes to avoid this ugliness....

Reply:
There's a reason your IDE has code-generators for this! It's write-only code.